### PR TITLE
[Samples] Add 11.qnamaker bot sample

### PR DESCRIPTION
**Pull request by matiasroldan6**
Fixes MS864

## Description
We are adding the sample 11.qnamaker in Java taking into account the C# [11.qnamaker bot](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/11.qnamaker) present in [BotBuilder-Samples](https://github.com/microsoft/BotBuilder-Samples) repository.

## Specific Changes
  - Generate main structure of the sample
  - Add `resources` folder with sample questions tsv
  - Add `deploymentTemplates` folder
  - Add root files
  - Add `application.properties`
  - Add LICENSE document
  - Add README document to describe the steps to run the sample

## Testing
Current behavior of the Java migrated sample
image

### CODE REVIEW

File: samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.microsoft.bot.ai.qna.QnAMakerEndpoint;
+import com.microsoft.bot.ai.qna.QnAMakerOptions;
+import com.microsoft.bot.builder.ActivityHandler;
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.integration.Configuration;
+import org.slf4j.LoggerFactory;
+import java.util.concurrent.CompletableFuture;
+
+public class QnABot extends ActivityHandler {
```
Comment (Batta32) [inline samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java:16]: Can we move the entire class into a `bots` folder just like [C#](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/csharp_dotnetcore/11.qnamaker/Bots)?

File: samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
+        QnAMakerEndpoint qnAMakerEndpoint = new QnAMakerEndpoint() {
+            {
+                setKnowledgeBaseId(configuration.getProperty("QnAKnowledgebaseId"));
+                setEndpointKey(configuration.getProperty("QnAEndpointKey"));
+                setHost(configuration.getProperty("QnAEndpointHostName"));
+            }};
```
Comment (Batta32) [inline samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java:30]:
```suggestion
        QnAMakerEndpoint qnAMakerEndpoint = new QnAMakerEndpoint() {
            {
                setKnowledgeBaseId(configuration.getProperty("QnAKnowledgebaseId"));
                setEndpointKey(configuration.getProperty("QnAEndpointKey"));
                setHost(configuration.getProperty("QnAEndpointHostName"));
            }
        };
```

File: samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        QnAMaker qnaMaker = new QnAMaker(qnAMakerEndpoint, null);
+
+        LoggerFactory.getLogger(QnABot.class).info("Calling QnA Maker");
+
+        QnAMakerOptions options = new QnAMakerOptions() {{
+            setTop(1);
+        }};
```
Comment (Batta32) [inline samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java:38]:
```suggestion
        QnAMakerOptions options = new QnAMakerOptions() {
            {
                setTop(1);
            }
        };
```

**Review by Batta32 — CHANGES_REQUESTED** [review]
After applying these changes, do a new end-to-end tests. Overall it's ok!

File: samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java (hunk 2 above)
Comment (matiasroldan6) [reply samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java:30]: thought the syntax was [like that](https://github.com/southworks/botbuilder-java/pull/61#discussion_r580414135). I'll change it here and in the other portion 👍

File: samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        // The actual call to the QnA Maker service.
+        qnaMaker.getAnswers(turnContext, options).thenAccept(response -> {
+            if (response != null && response.length > 0) {
+                turnContext.sendActivity(MessageFactory.text(response[0].getAnswer())).thenApply(sendresult -> null);
```
Comment (fran893) [inline samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java:43]: sendresult -> sendResult

**Review by fran893 — CHANGES_REQUESTED** [review]

File: samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            }
+            else {
+                turnContext.sendActivity(MessageFactory.text("No QnA Maker answers were found.")).thenApply(sendresult -> null);
```
Comment (fran893) [inline samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java:46]: Same as above

**Review by fran893 — CHANGES_REQUESTED** [review]

File: samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java (hunk 1 above)
Comment (matiasroldan6) [reply samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java:16]: I didn't see the class being inside a bots folder in [multilingual java](https://github.com/microsoft/botbuilder-java/tree/main/samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual) or the other samples 🤔. I will move it though, np

Comment (Batta32) [reply samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java:16]: Ack, my bad. I double checked and you are right. Don't change this

**Review by fran893 — CHANGES_REQUESTED** [review]
The QnABots.java should not be in the `bots` folder, it should be as it was originally

File: samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java (hunk 1 above)
Comment (matiasroldan6) [reply samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java]: yikes, this never refreshed :(. Guess I can go to a previous commit

**Review by fran893 — APPROVED** [review]

**Review by Batta32 — APPROVED** [review]
LGTM. This will be promoted tomorrow

**Comment by Batta32** [comment]
Merged in MS

**Pull request closed without merging**